roles/generate_manifests/tasks/main.yml

```diff
@@ -27,6 +27,20 @@
       }}"
   when: use_local_mirror_registry  | bool
 
+- name: Update api_vips variable when extra_api_vip is defined
+  ansible.builtin.set_fact:
+    api_vips: "{{ api_vips + [extra_api_vip] }}"
+  when:
+    - extra_api_vip is defined
+    - extra_api_vip | length > 0
```

Comment:
You can do the check for length in the inventory validation. That way simplify your check here. 

This would allow you to set `api_vips` in a relativity  simple jinja template within defaults like we do with dns for the assisted installer deployment [here](https://github.com/redhat-partner-solutions/crucible/blob/main/roles/setup_assisted_installer/defaults/main.yml#L50C1-L50C79). Allowing you to just use `api_vips` as if was natural. Just a suggestion if you would prefer to have it explicitly in tasks that's also fine.

Reply:
Thanks, I refined the validations with your suggestions and looks simpler this way. I launched a test to confirm this is non breaking current clusters without dualstack and looks good, but I still have pending to test again in a baremetal cluster with dualstack. I'll keep you posted.